include/lsp-plug.in/dsp-units/sampling/Sample.h

```diff
@@ -176,6 +176,15 @@ namespace lsp
                  */
                 bool resize(size_t channels, size_t max_length, size_t length = 0);
 
+                /** Stretch sample
+                 *
+                 * @param channels number of channels
+                 * @param stretch_samples samples to stretch
+                 * @param chunk_size chunk size multiplier
+                 * @return true if data was successfuly stretched
+                 */
+                bool stretch(size_t channels, int stretch_samples, size_t chunk_size);
```

Comment:
I think it could be much more flexible to define the range to stretch instead of stretching the full sample by parameters like `start` and `end` computed in samples. The simplified overloaded `stretch()` method can just set up `start` to `0` and `end` to `samples()` or `nLength`.

Reply:
I don't really get why the `start` param should be needed. Could you elaborate a bit further?